src/proxy/Map.ts

```diff
@@ -56,4 +124,269 @@ export class Map<K, V> extends BaseProxy implements IMap<K, V> {
     isEmpty(): Q.Promise<boolean> {
         return this.encodeInvokeOnRandomTarget<boolean>(MapIsEmptyCodec);
     }
+
+    getAll(keys: K[]): Q.Promise<any[]> {
+        var partitionService = this.client.getPartitionService();
+        var partitionsToKeys: {[id: string]: any} = {};
+        var key: K;
```

Comment:
I looked at it but I do not see an easy way to extract these operations to a common method. If you look closely, putAll operation is aware that it has pairs and it should use first element of the pair as key. Whereas, getAll opearation does not have key value pairs and uses elements of the array as keys directly. 

Also putAll has to do `partitionsToKeys[pId].push({key: keyData, val: this.toData(pair[1])})` different than getAll: `partitionsToKeys[pId].push(keyData);`.
I
 appreciate trying to re-use code but I do not see an easy way to achieve it in here.